services/spar/src/Spar/Scim/User.hs

```diff
@@ -643,11 +636,22 @@ getOrCreateScimUser stiTeam brigUser = do
       ssoIdentity' <- do
         -- TODO: If user is not an SSO User; @ssoIdentity'@ is Nothing
         -- Hence; we should only set managedByScim if this _succeeds_
+        -- and only lookup idp conditionally.
         getSSOIdentity' brigUser'
-      externalId <- toExternalId' ssoIdentity'
+      samlIdentity <- do
```

Comment:
Why this new code? what was wrong with `toExternalId'` ? It's not clear from context; not commit message, why this code was added to me